### PR TITLE
Add parameter for adjust current sign in battery plugin

### LINKED ISSUE #2685: Discrepancy in battery current sign between gz and ROS 2 for discharging state
## Environment
* OS Version: Ubuntu 24.04
* binary build version: 8.6.0

<details>

When using a ROS 2 bridge to convert battery state from gz to `sensor_msgs/BatteryState` in ROS 2, 
`sensor_msgs/BatteryState` expects the `current` to have a negative value during discharge.
However, the `gz-sim-linearbatteryplugin-system` plugin does not behave this way.
Is it correct for the `current` in the gz message to remain positive even during discharge?

</details>

## Description
* Expected behavior: 

current value is negative value.

* Actual behavior: 

current value is positive value.

## Steps to reproduce

1.
2.
3.

## Output

```
header:
  stamp:
    sec: 47
    nanosec: 885000000
  frame_id: ''
voltage: 22.915739059448242
temperature: 0.0
current: 21.886171340942383
charge: 1.7268285751342773
capacity: 2.0
design_capacity: .nan
percentage: 86.3414306640625
power_supply_status: 2
power_supply_health: 0
power_supply_technology: 0
present: true
cell_voltage: []
cell_temperature: []
location: ''
serial_number: '
```

### PR DESCRIPTION
**Pull request by Tacha-S**
# 🦟 Bug fix

Fixes #2685 

## Summary

A parameter has been added to adjust the sign of the battery current value to comply with ROS conventions.
By default, it is set to `false`, so the existing behavior remains unchanged.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/systems/battery_plugin/LinearBatteryPlugin.cc [hunk 1]
```diff
@@ -178,6 +178,9 @@ class gz::sim::systems::LinearBatteryPluginPrivate
 
   /// \brief Initial power load set trough config
   public: double initialPowerLoad = 0.0;
+
+  /// \brief Adjusts the sign of the current to align with ROS conventions.
+  public: bool adjustCurrentSignForROS{false};
```
Comment (ahcorde) [inline src/systems/battery_plugin/LinearBatteryPlugin.cc:183]: Do you mind to use a different value for the parameter ? We try not to include any ROS related stuff here.

Maybe something like `invertSignValues` ? But I open to suggestions

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: src/systems/battery_plugin/LinearBatteryPlugin.cc (hunk 1 above)
Comment (ahcorde) [reply src/systems/battery_plugin/LinearBatteryPlugin.cc:183]: go ahead with `invertCurrentSign` :ok_hand:

**Review by azeey — COMMENTED** [review]
Wouldn't this be easier to flip the sign only when the data is being published (PostUpdate) instead of adding extra logic at various places?

**Comment by Tacha-S** [comment]
That's better. I’ve done it.

**Review by azeey — APPROVED** [review]
Thanks for the contribution!

**Comment by ahcorde** [comment]
https://github.com/Mergifyio backport gz-sim8 ign-gazebo6

**Pull request merged**